### PR TITLE
normalize all paths

### LINKED ISSUE #6: Deal with non-html pages
At the moment, all URL's are (naively) saved as `index.html` files in a directory:

https://github.com/spatie/laravel-export/blob/e61cdd7d6e87ae467b6ed2f08b8abab45b30775c/src/ExportCrawlObserver.php#L36

However, if I'd add an entry point to an RSS feed or other non-html resource, this would break.

### PR DESCRIPTION
**Pull request by Gummibeer**
fixes #6 
extends #29

### CODE REVIEW

**Comment by Gummibeer** [comment]
One thing I ask myself and would like to discuss is if we should/could support URLs? Possibly with a new `urls()` method which only does `Str::replaceFirst($this->urlGenerator->to('/'), '', $url)`.
With this it would be possible to do `$exporter->urls([ route('home') ])`. I would be fine with the limitation that it has to be an app URL and not any foreign one.
And a second idea is to support single strings in the `paths()`/`urls()` methods as argument.
`$exporter->urls(route('home'))`. This would make it easier if anyone wants to add paths in a loop - saves the surrounding array.

**Comment by Gummibeer** [comment]
I've added https://github.com/spatie/laravel-export/pull/31/commits/a5071e80328dfcc1ab2fcfbf7bffa83dfd4a7640 to use the configured base URL for the fake request. This is passed through to all the helper functions. Otherwise it has used `http://localhost` all the time.

File: src/Traits/NormalizesPath.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Spatie\Export\Traits;
+
+use Illuminate\Support\Str;
+
+trait NormalizesPath
```
Comment (sebastiandedeyne) [inline src/Traits/NormalizesPath.php]: Could this be renamed to `normalizedPaths`?

**Comment by sebastiandedeyne** [comment]
> One thing I ask myself and would like to discuss…

Both ideas sound good!

File: src/Traits/NormalizesPath.php (hunk 1 above)
Comment (Gummibeer) [reply src/Traits/NormalizesPath.php]: Wasn't the rule that traits are named by the behavior thy add?
But for sure I can rename it.

**Comment by Gummibeer** [comment]
@sebastiandedeyne done all three things and also refactored the service provider a bit.
The exporter is now a simple singleton (without callback). So you don't have to resolve it's dependencies.
The default configuration is done during boot now.

File: src/Exporter.php [hunk 2]
```diff
@@ -47,13 +53,26 @@ public function crawl(bool $crawl): self
         return $this;
     }
 
-    public function paths(array $paths): self
+    public function paths(...$paths): self
     {
+        $paths = is_array($paths[0]) ? $paths[0] : $paths;
+
         $this->paths = array_merge($this->paths, $paths);
```
Comment (nuernbergerA) [inline src/Exporter.php]: Maybe shorten it
```php
$this->paths = array_merge($this->paths, Arr::wrap($paths));
```

File: src/Exporter.php [hunk 3]
```diff
@@ -47,13 +53,26 @@ public function crawl(bool $crawl): self
...
         return $this;
     }
 
+    public function urls(...$urls): self
+    {
+        $urls = is_array($urls[0]) ? $urls[0] : $urls;
+
+        $this->paths(array_map(function (string $url): string {
+            return Str::replaceFirst($this->urlGenerator->to('/'), '', $url);
+        }, $urls));
```
Comment (nuernbergerA) [inline src/Exporter.php]: Maybe shorten it
```php
$this->paths(array_map(function (string $url): string {
     return Str::replaceFirst($this->urlGenerator->to('/'), '', $url);
}, Arr::wrap($urls)));
```

Comment (nuernbergerA) [reply src/Exporter.php]: Why is it `$this->paths(array_map(...))` instead of `$this->paths = array_map(...)` ?

File: src/Exporter.php (hunk 2 above)
Comment (Gummibeer) [reply src/Exporter.php]: `Arr::wrap()` does something different. `is_array($paths[0]) ? $paths[0] : $paths` takes the first argument as array or all arguments as array.
```php
paths(['path1', 'path2']);

paths('path1', 'path2');
```

File: src/Exporter.php (hunk 3 above)
Comment (Gummibeer) [reply src/Exporter.php]: These methods aren't assignment methods instead they add values.

**Comment by Gummibeer** [comment]
@sebastiandedeyne this commit https://github.com/spatie/laravel-export/pull/31/commits/a5071e80328dfcc1ab2fcfbf7bffa83dfd4a7640 is very urgent because I always have to adjust it in vendor folder to do an export and netlify deployment.

File: src/Traits/NormalizesPath.php (hunk 1 above)
Comment (sebastiandedeyne) [reply src/Traits/NormalizesPath.php]: Sorry, apparently I completely derped while typing this comment. What I meant was `NormalizesPaths`. I'll change it myself after merging in a sec :)

**Comment by sebastiandedeyne** [comment]
Thanks!

**Pull request merged**